src/carExample/Car.java

```diff
@@ -1,172 +1,172 @@
+               int maxNumberOfPassengers, int actualNumberOfPassengers, int currentSpeed) {
+        this.productionDate = productionDate;
+        this.engineType = engineType;
+        this.maxSpeed = maxSpeed;
+        this.timeToHundreetKPH = timeToHundreetKPH;
+        this.maxNumberOfPassengers = maxNumberOfPassengers;
+        this.actualNumberOfPassengers = actualNumberOfPassengers;
+        this.currentSpeed = currentSpeed;
+    }
+
+    public void setCurrentSpeed(int speed) {
+        this.currentSpeed = speed;
+    }
+
+    public void addPassenger() {
+        if (actualNumberOfPassengers < maxNumberOfPassengers) {
+            actualNumberOfPassengers++;
+        } else {
+            System.out.println("Car is already full");
+        }
+    }
+
+    public void removePassenger() {
+        if (actualNumberOfPassengers > 0) {
+            actualNumberOfPassengers--;
+        } else {
+            System.out.println("Car is already empty");
+        }
+    }
+
+    public void removeAllPassengers() {
+        if (actualNumberOfPassengers > 0) {
+            actualNumberOfPassengers = 0;
+        } else {
+            System.out.println("Car is already empty");
+        }
+    }
+
+    public CarDoors getCarDoors(int i) {
+        if (i <= doors.size()) {
+            return doors.get(i);
+        } else {
+            System.out.println("Wrong index");
+            return doors.get(doors.size());
+        }
+    }
+
+    public CarWheels getCarWeels(int i) {
+        if (i <= wheels.size()) {
+            return wheels.get(i);
+        } else {
+            System.out.println("Wrong index");
+            return wheels.get(wheels.size());
+        }
+    }
+
+    public void removeAllWeels() {
+        for (CarWheels weel : wheels) {
+            weel = null;
+        }
+    }
+
+    public void installNewWeels(int numberToInstall) {
+        for (int i = 0; i < numberToInstall; i++) {
+            wheels.add(new CarWheels(0.5));
+        }
+    }
+
+    public void calculateActualMaxSpeed() {
+        if (actualNumberOfPassengers == 0) {
+            System.out.println("There is nobody in the car, speed is 0");
+            return;
+        }
+        double worstWheel = 1;
+        for (CarWheels wheel : wheels) {
+            if (wheel.getWheelState() < worstWheel) {
+                worstWheel = wheel.getWheelState();
+            }
+        }
+        System.out.println("Actual maximum speed is " + maxSpeed * worstWheel + " kmph");
+    }
+
+    //set the default set of wheels
+    public void setTheDefoltSetOfWheels() {
+        for (int wheel = 0; wheel < 5; wheel++){
+            wheels.add(new CarWheels(0.5));
+        }
+    }
+
+    //set the default set of doors
+    public void setTheDefoltSetOfDoors() {
+        for (int door =0; door < 5; door++) {
+            doors.add(new CarDoors(false, false));
+        }
+
+    }
+
+    public void printInfoAboutCar() {

```

Comment:
better override toString for that purpose

Reply:
I made both options

Reply:
> I made both options

ok, but in class Car I didnt see overriding method toString()

Reply:
yes indeed in this class forgot to override, corrected now sending